### ISSUE TITLE
:Files is un-naturally slow the first time

### ISSUE THREAD

**Issue by RedDocMD** [junegunn/fzf.vim#1332]
- [x] I have fzf 0.23.0 or above
- [x] I have read through https://github.com/junegunn/fzf.vim/blob/master/README.md
- [x] I have read through https://github.com/junegunn/fzf/blob/master/README-VIM.md
- [x] I have read through the manual page of fzf (`man fzf`)
- [x] I have searched through the existing issues

Off late, (for the past 2 weeks), the `:Files` command takes much longer than usual to appear for the first time. I am using `fzf 0.27.2 (e086f0b)` and `neovim v0.4.4`. Searching after the pop-up window appears is fast, subsequent invocations of `:Files` is fast. The problem occurs only when I open Neovim for the first time and invoke `:Files`. 
I have benchmarked my SSD, and its performance seems to be about the same (its also a new one, only about 6 months old).
Also, fzf works just as fast as usual when I do history search in the fish shell.

**Comment by junegunn** [comment]
I can't reproduce the problem. Can you check if you still have the issue with the following minimal init.vim?

```sh
nvim -Nu <(curl https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)
```

(https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)

**Comment by RedDocMD** [comment]
Yep, the problem still persists.

https://user-images.githubusercontent.com/54888685/134462254-48ed6993-a0b1-4be4-a1a5-3c8689a6d10c.mp4

I issue the `:Files` command at 0:14 and the pop-up appears at 0:17.

**Comment by junegunn** [comment]
Do you have the same problem on regular Vim?

```sh
\vim -Nu <(curl https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)
```

**Comment by RedDocMD** [comment]
> Do you have the same problem on regular Vim?
> 
> ```shell
> \vim -Nu <(curl https://gist.githubusercontent.com/junegunn/6936bf79fedd3a079aeb1dd2f3c81ef5/raw)
> ```

Yes, regular Vim (v8.2) gives the exact same issue (and almost exactly the same delay).

**Comment by junegunn** [comment]
Strange. Does `:echo system(fzf#exec() .. ' --version')` return immediately, or you see the same delay?

**Comment by RedDocMD** [comment]
Yup, the command takes about 3 seconds to execute the first time and less than 1 second subsequently.
So the delay is present here as well.

**Comment by junegunn** [comment]
I would check if running the same command (i.e. the output of `:echo fzf#exec() .. ' --version'`) outside of Vim or Neovim on a fresh shell gives the same delay.

**Comment by RedDocMD** [comment]
You mean running `fzf --version` on the shell?
That gives no delay. (Output  is `0.27.2 (e086f0b)`)

**Comment by junegunn** [comment]
Maybe you have a slow `$SHELL`? Vim will use it to run a command.

```sh
# Still slow?
SHELL=sh nvim
```

**Comment by RedDocMD** [comment]
I reckon `fish` is not a slow shell

**Comment by faisal-shah** [comment]
Maybe there is something wrong with the non-interactive shell startup? I think bash uses $BASH_ENV as non-interactive startup. Do you have something similar in fish?
